embedded-graphics/src/primitives/arc.rs

```diff
@@ -0,0 +1,634 @@
+
+        let iter = DistanceIterator::new(arc);
+
+        Self {
+            iter,
+            outer_threshold,
+            inner_threshold,
+        }
+    }
+}
+
+impl Iterator for Points {
+    type Item = Point;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        for (point, distance) in &mut self.iter {
+            if distance < self.outer_threshold && distance >= self.inner_threshold {
+                return Some(point);
+            }
+        }
+
+        None
+    }
+}
+
+/// Pixel iterator for each pixel in the arc border
+#[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
+pub struct StyledArcIterator<C>
+where
+    C: PixelColor,
+{
+    iter: DistanceIterator,
+
+    outer_threshold: u32,
+    inner_threshold: u32,
+
+    color: Option<C>,
+}
+
+impl<C> StyledArcIterator<C>
+where
+    C: PixelColor,
+{
+    fn new(styled: &Styled<Arc, PrimitiveStyle<C>>) -> Self {
+        let Styled { primitive, style } = styled;
+
+        let stroke_area = primitive.expand(style.outside_stroke_width());
+        let fill_area = primitive.shrink(style.inside_stroke_width());
+
+        let inner_threshold = circle::diameter_to_threshold(fill_area.diameter);
+        let outer_threshold = circle::diameter_to_threshold(stroke_area.diameter);
+
+        let iter = if !styled.style.is_transparent() {
+            DistanceIterator::new(&stroke_area)
+        } else {
+            DistanceIterator::empty()
+        };
+
+        Self {
+            iter,
+            outer_threshold,
+            inner_threshold,
+            color: styled.style.stroke_color,
+        }
+    }
+}
+
+impl<C> Iterator for StyledArcIterator<C>
+where
+    C: PixelColor,
+{
+    type Item = Pixel<C>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        for (point, distance) in &mut self.iter {
+            if distance < self.outer_threshold && distance >= self.inner_threshold {
+                if let Some(color) = self.color {
+                    return Some(Pixel(point, color));
+                }
+            }
+        }
+
+        None
+    }
+}
+
+impl<'a, C: 'a> Drawable<C> for &Styled<Arc, PrimitiveStyle<C>>
+where
+    C: PixelColor,
+{
+    fn draw<D: DrawTarget<C>>(self, display: &mut D) -> Result<(), D::Error> {
+        display.draw_arc(self)
+    }
+}
+
+impl<'a, C> IntoIterator for &'a Styled<Arc, PrimitiveStyle<C>>
+where
+    C: PixelColor,
+{
+    type Item = Pixel<C>;
+    type IntoIter = StyledArcIterator<C>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        StyledArcIterator::new(self)
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
+struct PlaneSector {
+    line_a: LinearEquation,
+    line_b: LinearEquation,
+    draw_above_a: bool,
+    draw_above_b: bool,
+    sweep: Angle,
+}
+
+impl PlaneSector {
+    fn new(arc: &Arc) -> Self {
+        let angle_start = arc.angle_start.normalize_from(-90.0.deg());
+        let angle_end = arc.angle_end().normalize_from(-90.0.deg());
```

Comment:
Why does this subtract 90 degree from the angle?

Reply:
First note that it doesn't subtract 90 degree from the angle, it selects the normalization starting point to -90 rather than 0.

Basically, to decide whether we should draw on the part of the plane which is below or above line_a (variable draw_above_a), we need to know if the angle is either between -90 and 90, or if it is between 90 and 270. To do this, the easiest way is to normalize the angle between -90 and 270 rather than 0 to 360.